### ISSUE TITLE
`connections_cft` test CI failures

### ISSUE THREAD

**Issue by jumaffre** [microsoft/CCF#2645]
https://dev.azure.com/MSRC-CCF/CCF/_build/results?buildId=26648&view=logs&j=ee9484ec-b476-577b-aa0e-d5ddfcbcd149&t=8e7d12cf-69d2-5704-84b7-b6fd04033d7b&l=21650

```
64: 13:53:37.193 | INFO     | __main__:get_session_metrics:27 - [0||] GET /node/metrics
64: 13:53:37.193 | INFO     | __main__:get_session_metrics:27 - 503 Service is currently busy and unable to serve new connections
64: 13:53:37.193 | INFO     | infra.network:network:1056 - Stopping network
64: 13:53:37.193 | INFO     | infra.node:stop:298 - Stopping node 0
64: 13:53:37.194 | INFO     | infra.remote:stop:489 - [127.21.147.28] closing
64: 13:53:37.358 | INFO     | infra.node:stop:298 - Stopping node 1
64: 13:53:37.359 | INFO     | infra.remote:stop:489 - [127.150.25.47] closing
64: 13:53:37.524 | INFO     | infra.node:stop:298 - Stopping node 2
64: 13:53:37.524 | INFO     | infra.remote:stop:489 - [127.10.238.72] closing
64: 13:53:37.689 | INFO     | infra.network:stop_all_nodes:534 - All nodes stopped
64: Traceback (most recent call last):
64:   File "/mnt/build/2/s/tests/connections.py", line 211, in <module>
64:     run(args)
64:   File "/mnt/build/2/s/tests/connections.py", line 180, in run
64:     final_metrics = get_session_metrics(primary)
64:   File "/mnt/build/2/s/tests/connections.py", line 28, in get_session_metrics
64:     assert r.status_code == http.HTTPStatus.OK, r.status_code
64: AssertionError: 503
```

Note that the test has also been consistently failing on the multi-threaded CI: https://dev.azure.com/MSRC-CCF/CCF/_build?definitionId=13

**Comment by achamayou** [comment]
`/node/metrics' is called perhaps too early for all the fds to wind down, there should be a retry with a delay here.

**Comment by achamayou** [comment]
@eddyashton new one after the most recent fix:

https://dev.azure.com/MSRC-CCF/CCF/_build/results?buildId=27791&view=logs&j=a45bc589-b130-5dc2-b16b-03ef91a336e9&t=643c4120-e5c7-597b-5ce9-cf0a315b4855
```
70: Traceback (most recent call last):
70:   File "/mnt/build/3/s/tests/connections.py", line 211, in <module>
70:     run(args)
70:   File "/mnt/build/3/s/tests/connections.py", line 203, in run
70:     raise RuntimeError("Expected a fatal crash and saw none!")
70: RuntimeError: Expected a fatal crash and saw none!
63/76 Test #70: connections_cft ...................***Failed   64.61 sec
```

**Comment by eddyashton** [comment]
So what happens here is, we lower the fd limit on the node (below even the soft session cap), and then open enough parallel sessions to exhaust the fds. This isn't a situation we can handle politely, and it generally causes us to crash. But it doesn't _consistently_ cause a crash. We crash if we try to do certain kinds of ledger operations while the fds are exhausted (those are deliberately crashing rather than robust to this, because we don't think they should ever happen in practice and want to hear loudly if they do). But I can't see how to guarantee that those operations happen - I thought we were flushing enough transactions through to trigger them, but it seems sometimes we continue to write safely to existing ledger fds, and only fail on snapshot writes (which _are_ deliberately non-critical and robust, so don't cause a node failure).

Long story short, I think we just need to remove that assertion. If we exhaust a dangerously low fd limit, we _might_ crash a node but won't necessarily. Asserting that we always do was an attempt at formalising the behaviour, but this is too strict. So I'll weaken that assertion.
